File: android-samples/poslink/EposPosLinkExample/app/src/main/java/com/example/eposappexample/poslink/transactions/TransactionRecord.kt
package com.example.eposappexample.poslink.transactions

import com.teya.unifiedepossdk.models.TransactionType

data class TransactionRecord(
    val id: String,
    val type: TransactionType,
    val isSuccess: Boolean,
    val statusLabel: String,
    val amountMinor: Int,
    val currency: String,
    val gatewayPaymentId: String?,
    val timestamp: Long,
    val isRefunded: Boolean = false,
) {

    // a referenced refund needs the original payment's gatewayPaymentId, so it must be present
    val isRefundable: Boolean
        get() = type == TransactionType.Payment && isSuccess && !isRefunded && gatewayPaymentId != null
}

File: windows-samples/poslink/EposPosLinkExample/EposPosLinkExample/Models/TransactionRecord.cs
namespace EposPosLinkExample.Models
{
    public enum TransactionType { Payment, Refund }

    public record TransactionRecord(
        string Id,
        TransactionType Type,
        bool IsSuccess,
        int AmountMinor,
        string Currency,
        string? GatewayPaymentId,
        long Timestamp)
    {
        public bool IsRefunded { get; init; }

        public bool IsRefundable => Type == TransactionType.Payment && IsSuccess && !IsRefunded && GatewayPaymentId != null;
    }
}
